intel_extension_for_transformers/neural_chat/tools/evaluation/context_utils.py

```diff
@@ -0,0 +1,246 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unicodedata
+import pandas as pd
+import re, json
+from langchain.document_loaders import UnstructuredMarkdownLoader
+from docx import Document as DDocument
+from bs4 import BeautifulSoup
+import fitz
+import easyocr
+from PIL import Image
+import numpy as np
+import io
+
+def uni_pro(text):
+    """Check if the character is ASCII or falls in the category of non-spacing marks."""
+    normalized_text = unicodedata.normalize('NFKD', text)
+    filtered_text = ''
+    for char in normalized_text:
+        if ord(char) < 128 or unicodedata.category(char) == 'Mn':
+            filtered_text += char
+    return filtered_text
+
+
+def read_pdf(pdf_path):
```

Comment:
shall we put such utilities to some common place? @XuhuiRen @Liangyx2

Reply:
Yes, should reuse the code in retrieval. Updated.